65black/so-lite-react
# #162832595 consume put endpoint for answers

#### What does this PR do?
This PR adds the ability for a logged in user to edit their already posted answers. It also adds the ability for a logged in user to choose an answer as preferred for their question.

#### Description of Task to be completed?
- add ability to edit answers
- add ability to set answer as preferred
- add tests for new features

#### How should this be manually tested?
- Clone the repo, run `npm install` and run `npm start`
- Log in to your account #3 
- Navigate to `localhost:7000/questions` and click on any question
- Choose answer you own and click on edit. Edit the answer as required and hit submit
- The page should refresh and you should see the edited version of your answer

Testing selection of preferred answers

- Navigate to `localhost:7000/profile` and click on any question
- Select your preferred answer by clicking on the accept button below the answer
- The page should refresh and you should see a light-green background color on the answer

NOTE: Once you select an answer as preferred, you can't choose any other one.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#162832595

#### Screenshots (if appropriate)
<img width="1440" alt="screenshot 2018-12-30 at 9 55 54 am" src="https://user-images.githubusercontent.com/36507972/50545669-253d4980-0c19-11e9-994b-aabad304a8fe.png">
<img width="1440" alt="screenshot 2018-12-30 at 9 55 16 am" src="https://user-images.githubusercontent.com/36507972/50545670-25d5e000-0c19-11e9-9dc2-39209d2f335b.png">

#### Questions:
N/A
